### PR TITLE
CircleCI Pipeline Parallelization

### PR DESCRIPTION
**Pull request by dfangl**
Test PR for pipeline parallelization, split directly from #5415 .

This PR splits the docker-tests into multiple runners, while NOT executing the modules in a predictive manner (modules can be shuffled, however the order within them should stay the same).

Due to this possible shuffling, many tests were changed to use unique names for the created services, to prevent state from previous tests influencing the currently running test.

TODOs:

- [x] rebase to master

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -266,17 +278,21 @@ jobs:
...
           command: docker load -i target/localstack-docker-image-<< parameters.platform >>.tar
       - run:
-          name: Run tests
+          name: Run integration tests
+          # circleci split returns newline separated list, so `tr` is necessary to prevent problems in the Makefile
           command: |
-            PYTEST_ARGS='--junitxml=target/reports/test-report-<< parameters.platform >>.xml' \
-            COVERAGE_FILE='target/coverage/.coverage.<< parameters.platform >>' \
+            TEST_FILES=$(circleci tests glob "tests/integration/**/test_*.py" | circleci tests split --split-by=timings | tr '\n' ' ')
+            PYTEST_ARGS="-o junit_family=legacy --junitxml=target/reports/test-report-<< parameters.platform >>-${CIRCLE_NODE_INDEX}.xml" \
```
Comment (dfangl) [inline .circleci/config.yml:293]: junit_family=legacy is necessary for filenames to be included in the resulting junit-style xml.
This is necessary for the timing split to work.

File: tests/integration/test_iam.py [hunk 2]
```diff
@@ -16,6 +16,7 @@ class TestIAMIntegrations(unittest.TestCase):
     def setUp(self):
         self.iam_client = aws_stack.create_external_boto_client("iam")
 
+    # TODO what does this test do?
```
Comment (alexrashed) [inline tests/integration/test_iam.py:19]: nit: Have you found out what it does? Does the TODO have to stay in the code? :D

**Review by alexrashed — APPROVED** [review]
This is really a great set of changes! The test stabilization is really quite a heavy lifting and was much overdue! Great to see the tests being cleaned up, the cross-platform emulation being removed, and the execution being parallelized!

File: tests/integration/test_iam.py (hunk 2 above)
Comment (dfangl) [reply tests/integration/test_iam.py:19]: It will stay for now, but might be removed in the future :D

**Pull request merged**